proto/lbm/token/v1/query.proto

```diff
@@ -11,125 +11,237 @@ option go_package = "github.com/line/lbm-sdk/x/token";
+// Since: finschia
+message QueryTokenClassesResponse {
+  // information of the token classes.
+  repeated TokenClass classes = 1 [(gogoproto.nullable) = false];
+  // pagination defines the pagination in the response.
   lbm.base.query.v1.PageResponse pagination = 2;
 }
 
-// QueryGrantsRequest is the request type for the Query/Grants RPC method
-message QueryGrantsRequest {
-  // class id associated with the token.
-  string class_id = 1;
-  string grantee  = 2;
+// QueryGrantRequest is the request type for the Query/Grant RPC method
+// Since: finschia
+message QueryGrantRequest {
+  // contract id associated with the token class.
+  string contract_id = 1;
+  // grantee which has permissions on the token class.
+  string grantee = 2;
+  // a permission given to the grantee.
+  string permission = 3;
+}
+
+// QueryGrantResponse is the response type for the Query/Grant RPC method
+// Since: finschia
+message QueryGrantResponse {
+  // if no grant found for the request, it would be nil and return an error.
+  Grant grant = 1;
+}
+
+// QueryGranteeGrantsRequest is the request type for the Query/GranteeGrants RPC method
+// Since: finschia
+message QueryGranteeGrantsRequest {
+  // contract id associated with the token class.
+  string contract_id = 1;
+  // grantee which has permissions on the token class.
+  string grantee = 2;
+
+  // pagination defines an optional pagination for the request.
+  lbm.base.query.v1.PageRequest pagination = 3;
 }
 
-// QueryGrantsResponse is the response type for the Query/Grants RPC method
-message QueryGrantsResponse {
+// QueryGranteeGrantsResponse is the response type for the Query/GranteeGrants RPC method
+// Since: finschia
+message QueryGranteeGrantsResponse {
+  // all the grants on the grantee.
   repeated Grant grants = 1 [(gogoproto.nullable) = false];
+  // pagination defines the pagination in the response.
+  lbm.base.query.v1.PageResponse pagination = 2;
 }
 
-// QueryApproveRequest is the request type for the Query/Approve RPC method
-message QueryApproveRequest {
-  // class id associated with the token.
-  string class_id = 1;
-  string proxy    = 2;
-  string approver = 3;
+// QueryAuthorizationRequest is the request type for the Query/Authorization RPC method
+// Since: finschia
+message QueryAuthorizationRequest {
+  // contract id associated with the token class.
+  string contract_id = 1;
+  // address of the operator which the authorization is granted to.
+  string operator = 2;
+  // address of the token holder which has approved the authorization.
+  string holder = 3;
 }
 
-// QueryApproveResponse is the response type for the Query/Approve RPC method
-message QueryApproveResponse {
-  Approve approve = 1;
+// QueryAuthorizationResponse is the response type for the Query/Authorization RPC method
+// Since: finschia
+message QueryAuthorizationResponse {
+  // if no authorization found for the request, it would be nil and return an error.
```

Comment:
> Could you describe what kind of error will be sent to clients?

The error would be `types/errors.ErrNotFound`.

> Such information is necessary for client developers to implement their clients.

I will consider documenting it on x/token and x/collection.

> And I wonder whether the error response can contain `QueryAuthorizationResponse`.
> You wrote "it would be nil and return an error", so it seems that the error response has the field `authorization`. Right?

My fault. The entire design of the responses should be fixed, because as you pointed out, on the error, the response could be nil and I don't need to fill out the field. So I will set the `nullable` option of the field to `false`, which means the field must have a sane value in every occasion (where the error is `nil`).